### ISSUE TITLE
Does BatchNormalization support 2D shape of `X` input

### ISSUE THREAD

**Issue by xhcao** [microsoft/onnxruntime#25230]
### Describe the issue

From  `BatchNormalization` spec, the input `X`'s dimensions are in the form of (N x C x D1 x D2 … Dn), the spec does not show that `D` part must be needed.
Run an BatchNormalization unit case with 2D shape of `X` input, the case could pass on CPU EP, and the source code only restrict the `X`'s shape size >= 1. https://github.com/microsoft/onnxruntime/blob/539e785f30eb34d8131fe8efc1f536f73446a0e4/onnxruntime/core/providers/cpu/nn/batch_norm_helper.h#L31
`BatchNormalization` is layout sensitive operator, https://github.com/microsoft/onnxruntime/blob/539e785f30eb34d8131fe8efc1f536f73446a0e4/onnxruntime/core/optimizer/transpose_optimization/onnx_transpose_optimization.cc#L3308, if layout is changed from nchw to nhwc, the unit case will meet the errors https://github.com/microsoft/onnxruntime/blob/539e785f30eb34d8131fe8efc1f536f73446a0e4/onnxruntime/core/graph/contrib_ops/nhwc_inference_context.h#L108 and https://github.com/microsoft/onnxruntime/blob/539e785f30eb34d8131fe8efc1f536f73446a0e4/onnxruntime/core/graph/contrib_ops/nhwc_inference_context.h#L86
The case will pass if running it on webgpu ep with DataLayout::NCHW, which is the same as cpu EP. But it will fail on webgpu ep with DataLayout::NHWC.
It seems that there are some conflicts of `X` shape requirement here.

### To reproduce

Add an unit case in https://github.com/microsoft/onnxruntime/blob/main/onnxruntime/test/providers/cpu/nn/batch_norm_op_test.cc
TEST(BatchNormTest, PositiveTestCase_2D) {
  // This input was taken from the SpatialBN_1.pb, SpatialBN_1_input.pb and SpatialBN_1_output.pb files.
  vector<float> X{0.329876f, -0.287158f, -0.411425f, 0.473621f, 0.18156f, -0.170596f, -0.329516f};
  vector<float> scale{0.589433f, 0.589433f, 0.589433f, 0.589433f, 0.589433f, 0.589433f, 0.589433f};
  vector<float> B{-0.384622f, -0.384622f, -0.384622f, -0.384622f, -0.384622f, -0.384622f, -0.384622f};
  vector<float> mean{-2.45673f, -2.45673f, -2.45673f, -2.45673f, -2.45673f, -2.45673f, -2.45673f};
  vector<float> var{1.37998f, 1.37998f, 1.37998f, 1.37998f, 1.37998f, 1.37998f, 1.37998f};

  InputDataMap input_data_map;
  input_data_map.insert({"X", X});
  input_data_map.insert({"scale", scale});
  input_data_map.insert({"B", B});
  input_data_map.insert({"mean", mean});
  input_data_map.insert({"var", var});

  InputShapesMap input_shapes_map;
  vector<int64_t> input_shape{1, 7};
  input_shapes_map.insert({"X", input_shape});
  input_shapes_map.insert({"scale", {7}});
  input_shapes_map.insert({"B", {7}});
  input_shapes_map.insert({"mean", {7}});
  input_shapes_map.insert({"var", {7}});

  auto expected_output = {1.01359f, 0.703983f, 0.641631f, 1.08571f, 0.939167f, 0.762469f, 0.682729f};
  float epsilon = 1e-05f;
  TestBatchNorm(input_data_map, input_shapes_map, epsilon, expected_output, input_shape);
}

### Platform

Windows

### OS Version

windows 10

### ONNX Runtime Installation

Built from Source

### ONNX Runtime Version or Commit ID

505b135cfa89445b760a0a5efda84ec78a6e777e

### ONNX Runtime API

C++

### Architecture

X64

### Execution Provider

Other / Unknown

**Comment by fdwr** [comment]
From [the ONNX spec](https://onnx.ai/onnx/operators/onnx__BatchNormalization.html), BatchNormalization should logically accept any rank >=1D:

- ND – [N,C,D1,D2,D3,D#…]
- 5D – [N,C,D,H,W]
- 4D – [N,C,H,W]
- 3D – [N,C,W]
- 2D – [N,C] (spatial dimensions treated as 1’s)
- 1D – [N] (C treated as 1, along with spatial dimensions)

> “The op also accepts single dimension input of size N in which case C is assumed to be 1”
> https://onnx.ai/onnx/operators/onnx__BatchNormalization.html

So if ORT applies an optimization which leads to errant validation, that’s a bug, because *optimizations* shouldn’t undermine the *logical behavior*, but what's unclear to me is whether it's the EP's responsibility to massage the tensor first to 3D (by appending/prepending 1’s to the shape) before it reaches [this point](https://github.com/microsoft/onnxruntime/blob/539e785f30eb34d8131fe8efc1f536f73446a0e4/onnxruntime/core/graph/contrib_ops/nhwc_inference_context.h#L109), or whether the other parts of ORT should handle it gracefully. 🤔

**Comment by xhcao** [comment]
@fdwr  I do not know the root cause why add a if-statement [here](https://github.com/microsoft/onnxruntime/blob/539e785f30eb34d8131fe8efc1f536f73446a0e4/onnxruntime/core/graph/contrib_ops/nhwc_inference_context.h#L109), Is it reasonable?
If 2D shape is acceptable, and EP could support `X`'s 2D shape, so webnn and EP have no responsibility to change 2D tensor to 3D.
I think [the function](https://github.com/microsoft/onnxruntime/blob/539e785f30eb34d8131fe8efc1f536f73446a0e4/onnxruntime/core/optimizer/layout_transformation/layout_transformation.cc#L96) could handle the issue, but I am not familiar with ort-core code, it needs a ort-core expert to handle this issue.
Thanks.

**Comment by fdwr** [comment]
> it needs a ort-core expert to handle this issue

Yep, which is not me 😅. @skottmckay should know.

**Comment by fdwr** [comment]
Or @fs-eire should know more, seeing he touched that line.

**Comment by fs-eire** [comment]
In my understanding, I think this is a bug. It may be debatable because the ONNX spec is not very clear about 2D case for BatchNormalization, but I agree that if it worked before, it should not fail just by applying the layout optimization.

**Comment by miaobin** [comment]
WebNN WPT conformance test also encountered a similar problem when testing `batchNormalization` operator.
Since we used `com.ms.internal.nhwc::BatchNormalization`, we encountered an error when testing a 2-D input: "Tensor must have at least 3 dimensions to convert between channels first and channels last." This error is reported on [L109](https://github.com/microsoft/onnxruntime/blob/539e785f30eb34d8131fe8efc1f536f73446a0e4/onnxruntime/core/graph/contrib_ops/nhwc_inference_context.h#L109).
